compiler-core/src/language_server/code_action.rs

```diff
@@ -8623,10 +8623,16 @@ impl<'a> ExtractFunction<'a> {
                 statements,
                 location: full_location,
             }) => {
-                let location = SrcSpan::new(
-                    statements.first().location().start,
-                    statements.last().location().end,
-                );
+                let location = statements
+                    .first()
+                    .location()
+                    .merge(&match &statements.last() {
+                        ast::Statement::Expression(expression) => expression.location(),
+                        ast::Statement::Assignment(assignment) => assignment.location,
+                        ast::Statement::Assert(assert) => assert.location,
+                        ast::Statement::Use(use_) => use_.call.location(),
+                    });
```

Comment:
This code is a bit cryptic, and it is in 3 places in this PR. Could we extract it to a new documented helper method please? 🙏

Reply:
Yeah I was thinking about it but couldn't come up with a good name, what do you think it should be called? `entire_location`?

Reply:
Ended up moving that to the location function, I think that's what makes the most sense for use statements